### DISCUSSION TITLE
Ideas to re-use the `references/` folder in other documentation

### DISCUSSION THREAD

**Discussion by choldgraf** [pradyunsg/furo#181] [Ideas] [answered]
I love [the reference section](https://pradyunsg.me/furo/reference/) of Furo. It both covers a lot of great content, is easier to navigate than many "kitchen sink" style references, and does a great job of highlighting both rST and MyST.

I liked it so much that I even tried to get it running on the Sphinx Book Theme, you can find a PR for that here: https://github.com/executablebooks/sphinx-book-theme/pull/353

That PR basically:

1. Downloads a ZIP of Furo's repository, and extracts all the files under `reference` into a directory in our docs
2. Installs `furo` and activates `furo.sphinxext` in our `conf.py`

I ran into some issues which made it hard (or impossible?) to re-use. Opening this issue to list a few ideas that came up:

- The main issue is that the output doesn't show up - whatever `furo-demo` outputs seems to be an empty container and I'm not sure why.
- In terms of the content itself, there are two cross-refs that are Furo-specific: `admonitions.md:177` and `images.md:60`
- **`furo-demo`** could be a useful extension to include in a more generalizable form for people that want to show multiple syntaxes for the same behavior. I wonder if an easy path forward would be to define a directive like `demo-markup` that would take its contents, display them as a `literal` block, and then also display the parsed contents just below so you see how that literal block renders. Then this could be combined with your `tabs` example and would be similarly flexible and a bit simpler than hard-coding rst/myst structure in `furo-demo`.

No idea if any of these ideas are actionable, but wanted to share them in case they're of interest to you :-)

**Comment by pradyunsg** [comment]
How relevant is this still? :)

**Reply by choldgraf** [reply]
IMO still relevant 🙂 though that's partially selfish because i mostly want to reuse them for the pydata and book themes haha. I bet other developers would find it useful though - i wonder if a standard set of reference pages could be used at sphinx-themes and then made easy to download the source files for themes that wanted to reuse them?

**Reply by choldgraf** [reply]
Basically, I think it'd be helpful if there were a single location I could `wget` that:

- Was self-contained
- Was well-organized
- Covered 95% of syntax use-cases in Sphinx
- Didn't have errors in builds
- Ideally would have whiz-bang features like some kinda tabbed content for syntax options, but IMO that is not strictly necessary :-)
- Was something that the community could improve and contribute to as a shared resource

**Reply by pradyunsg** [reply]
So... https://github.com/sphinx-themes/sphinx-themes.org/tree/master/sample-docs/kitchen-sink? ;)

**Reply by choldgraf** [reply]
mmm that is super close, but it has "known errors" in it, which is really annoying if you want to tell Sphinx to fail on errors (e.g. known-missing footnotes: https://github.com/sphinx-themes/sphinx-themes.org/blob/master/sample-docs/kitchen-sink/paragraph-markup.rst#footnotes). I know it's tricky because you *want* to see what the syntax looks like for missing footnotes, but I think that building docs with "fail on error" is somewhat common as well. Thus far, I haven't found a way to suppress errors like that in Sphinx

**Reply by pradyunsg** [reply]
Well, even Furo’s kitchen sink has that. Same for ReadTheDocs theme.

The only reason the reference doesn’t, is because it doesn’t try to exercise all the elements. :/

**Reply by pradyunsg** [reply]
Hmm.. how do you feel if I add a errors.rst file that you’ll need to remove from a toctree + delete, which has all of that stuff?

**Reply by choldgraf** [reply]
IMO that is totally fine. I think the annoying thing is having errors embedded in like line 273 of 546 so you have to remember to go fishing for them whenever you update 🙂

**Reply by choldgraf** [reply]
Think about it - it might also be helpful to have this as a dedicated file if only because "handling errors" are kind-of a special case anyway. Perhaps it'd be helpful from a user-facing perspective as well, not just for needy developers like myself?

**Reply by pradyunsg** [reply]
Yea. Some other weekend tho. :)

**Reply by choldgraf** [reply]
Would you be open to review a light PR that does this?

**Reply by choldgraf** [reply]
I decided to just manually check for "unexpected warnings" in our test suite, so for me the need to have a separate errors file is not as strong now. I'm happy to just consider this resolved (even though it's not an issue 🙂)

**Comment by pradyunsg** [comment]
@choldgraf Anything more actionable here? We have the same content being shared across a bunch of themes IIUC. Should I make this a part of sphinx-theme-builder somehow, to allow it to synchronise the kitchen sink by manually running a command?

**Reply by choldgraf** [reply]
I think this is pretty much done, and it is lovely IMO 🙂

Though I keep getting a weird auto summary error about kissing urllib stub files i haven't been able to debug..have you run into that?

**Reply by choldgraf** [reply]
I do really like the "synchronize by running a command" idea... I wouldn't make it too complex, maybe just a one liner to download all of the latest versions?

**Reply by pradyunsg** [reply]
Do you have links to the code you're using for this right now?

**Reply by choldgraf** [reply]
- our kitchen sink:  https://github.com/pydata/pydata-sphinx-theme/tree/main/docs/examples/kitchen-sink
- index file: https://github.com/pydata/pydata-sphinx-theme/blob/main/docs/examples/kitchen-sink/index.rst
- and example action warnings: https://github.com/pydata/pydata-sphinx-theme/runs/7591218373?check_suite_focus=true#step:6:58

**Reply by choldgraf** [reply]
Ah and the script to download it: https://github.com/pydata/pydata-sphinx-theme/blob/main/docs/scripts/update_kitchen_sink.py

**Comment by pradyunsg** [comment, marked as answer]
I'm gonna mark this as "answered", based on https://github.com/pradyunsg/furo/discussions/181#discussioncomment-3290784.
